Documentation: Clearer compatibility listing

raybejjani opened the pull request:
**Summary of changes**:
This is a start at recording all the various versions we need in order to provide specific functionality. It is intended to address #1873 but needs more data about other cases to be useful. It is incomplete and I/we need to fill in sections marked ?? at least

@tgraf aanm fixed up that particular message last week. Did you still mean to have the issue open? Is this PR too broad for this case?

@aanm Is there a listing for k8s NetworkPolicy API versions that's similar to 
https://kubernetes.io/docs/reference/workloads-18-19/ ? I couldn't find one.

Fixes: #1873 

```release-note
Documented kubernetes API Group usage in docs
cilium status returns which kubernetes API Groups are supported/used by the agent
```

raybejjani committed:
Documentation: Clearer k8s compatibility doc

Signed-off-by: Ray Bejjani <ray@covalent.io>

aanm commented on Documentation/compatibility.rst:
```diff
@@ -0,0 +1,84 @@
++-------------+-------------------------------------------------------+
+| 1.7         | * k8s NetworkPolicy extensions/v1beta1                |
+|             | * k8s NetworkPolicy networking.k8s.io/v1              |
+|             | Exclusively one of:                                   |
```
Remove the `Exclusively one of` and add `CiliumNetworkPolicy as a CustomResourceDefinition ` on all k8s versions after 1.6

aanm reviewed (changes requested):
>@aanm Is there a listing for k8s NetworkPolicy API versions that's similar to
>https://kubernetes.io/docs/reference/workloads-18-19/ ? I couldn't find one.

I'm not aware of any

tgraf commented on Documentation/compatibility.rst:
```diff
@@ -0,0 +1,84 @@
+kernels
+-------
+
+cilium requires kernel version ?? and newer.
```
Requirements are covered here:
http://docs.cilium.io/en/stable/install/#system-requirements

tgraf commented on Documentation/compatibility.rst:
```diff
@@ -0,0 +1,84 @@
++------------+-------------------------------------------------+
+| KVStore    | Version | cilium Features/Abilities             |
++============+=================================================+
+| etcd       | >??     | * Lock-free ID allocation             |
```
You can skip this, I'll add this to the docs together with the addition of the new kvstore code. This code has not been merged yet.

raybejjani committed:
cli: Fix cilum status IP allocations printing

The other fields use a tabwriter to allow tab alignment. The IP
addresses simply printed to stdout. This make all the text use this to
avoid ordering issues when flushing the tabwriter.

Signed-off-by: Ray Bejjani <ray@covalent.io>

raybejjani committed:
api: Add K8sStatus to StatusResponse

raybejjani committed:
cli: Add status field listing K8s APIs we use/support

Signed-off-by: Ray Bejjani <ray@covalent.io>

raybejjani committed:
agent: Daemon.DumpIPAM usage has no hidden params

DumpIPAM would return nil when a debug was enabled. Instead of hiding
this behaviour in the function, limiting it's usefulness, we now check
for that at the only callsite and handle debug mode there.
